### ISSUE TITLE
[core] implement extractor and shift to jsonl as internal format

### ISSUE THREAD

**Issue by rudolfix** [scale-vector/dlt#33]
Implement extractor that will replace ad hoc implementation in the `Pipeline`
* [ ] stream processing - never loads more than one iteration at a time into memory
* [ ] stores data in jsonl (as a consequence) - this will allow unpacker to follow the same pattern
* [ ] for deferred iterators uses multi threading pool
* [ ] generates `load_id` that is preserved until data is loaded
* [ ] may drop the event count from the file name
* [ ] optional: `extract_many` to extract many iterators in parallel
* [ ] moves whole folders to unpacker to be more atomic
